docs/rules/no-dupe-disjunctions.md

```diff
@@ -87,11 +88,78 @@ This option control what types of duplications will be reported. The possible va
 
   If your codebase contained many such partial regexes, then reporting all cases might yield cases that could not be identified as causing exponential backtracking.
 
-### `alwaysReportExponentialBacktracking: boolean`
+### `reportExponentialBacktracking`
 
-If set to `true`, then this rule will always report partial duplications that can cause exponential backtracking. This option is set to `true` by default.
+Partial duplications (overlaps) are only reported by `report: "all"` even though they sometimes cause exponential backtracking. This option will force the other `report` modes to also report partial duplications if the partial duplications are likely to cause exponential backtracking.
 
-Only set this option to `false` if you have some other mean to reliably detect exponential backtracking.
+- `reportExponentialBacktracking: "potential"` (_default_)
```

Comment:
Regarding the defaults for `reportExponentialBacktracking` and `reportUnreachable`:

Right now both are set to `"potential"` which will produce some false positives. I worry that users might get annoyed by the over-reporting and turn off the rule.

I think we should change the default for `reportUnreachable` to `"certain"`. It's not catastrophically bad to have some false negatives here.

As for `reportExponentialBacktracking`, I think `potential` is okay. Missing cases can be catastrophic, so I think it's okay to have some false positives.

Reply:
I agree that `reportUnreachable: "certain"` should be the default. I will change it.